idseq_dag/steps/run_srst2.py

```diff
@@ -0,0 +1,110 @@
+    See: https://github.com/katholt/srst2
+    '''
+    def run(self):
+        ''' Invoking srst2 '''
+        db_file_path = fetch_from_s3(self.additional_files["resist_gene_db"], self.output_dir_local)
+        min_cov = str(self.additional_attributes['min_cov'])
+        n_threads = str(self.additional_attributes['n_threads'])
+        if len(self.input_files_local[0]) == 2:
+            fwd_rd = self.input_files_local[0][0]
+            rev_rd = self.input_files_local[0][1]
+            # Rename to format srst2 expects
+            # TODO: Handle Fasta files
+            command.execute(f"ln -sf {fwd_rd} _R1_001.fastq.gz")
+            command.execute(f"ln -sf {rev_rd} _R2_001.fastq.gz")
+            srst2_params = [
+                'srst2', '--input_pe', '_R1_001.fastq.gz', '_R2_001.fastq.gz', '--forward', '_R1_001', '--reverse', '_R2_001',
+                '--min_coverage', min_cov,'--threads', n_threads, '--output', os.path.join(self.output_dir_local, 'output'),
+                '--log', '--gene_db', db_file_path
+            ]
+        else:
+            rd = self.input_files_local[0][0]
+            command.execute(f"ln -sf {rd} rd.fastq.gz")
+            srst2_params = [
+                'srst2', '--input_se', 'rd.fastq.gz', '--min_coverage', min_cov,'--threads', n_threads,
+                '--output', self.output_dir_local+'/output', '--log', '--gene_db', db_file_path
+            ]
+        command.execute(" ".join(srst2_params))
+        log = os.path.join(self.output_dir_local, 'output.log')
+        log_dest = self.output_files_local()[0]
+        results = os.path.join(self.output_dir_local, 'output__genes__ARGannot_r2__results.txt')
+        results_dest = self.output_files_local()[1]
+        PipelineStepRunSRST2.mv_to_dest(log, log_dest)
+        PipelineStepRunSRST2.mv_to_dest(results, results_dest) 
+        if not os.path.exists(os.path.join(self.output_dir_local, 'output__fullgenes__ARGannot_r2__results.txt')): 
+            open('empty_file.txt','r').close()
+            # TODO: Have more efficient command than cp
+            command.execute(f"cp empty_file.txt {self.output_files_local()[2]}")
+            command.execute(f"cp empty_file.txt {self.output_files_local()[3]}")
+            command.execute(f"cp empty_file.txt {self.output_files_local()[4]}")
+        else:
+            # Post processing of amr data
+            results_full = os.path.join(self.output_dir_local, 'output__fullgenes__ARGannot_r2__results.txt')
+            results_full_dest = self.output_files_local()[2]
+            PipelineStepRunSRST2.mv_to_dest(results_full, results_full_dest)
+            self.process_amr_results(results_full_dest)
+
+    # Inherited method
+    def count_reads(self):
+        pass
+
+    @staticmethod
+    def mv_to_dest(src, dest):
+        """Moves src to dest path. 
+           Handles case where src and dest file are named the same and a mv "same file" error is thrown."""
+        if src == dest:
```

Comment:
if src == dest, you don't need to do anything...
```
def mv_to_dest(src, dest):
    if src != dest: 
        command.execute(f"mv {src} {dest}")
```
Also, technically for this to work correctly in all cases, you want to compare the real paths rather than the `src`/`dest` strings. E.g. `/Users/meera/../meera/file.txt` is the same location as `/Users/meera/file.txt`. So you would use the comparison `if os.path.normpath(src) != os.path.normpath(dest)`. 

Update:
I just tried `os.path.rename` and it turns out that it correctly handles the case where source and destination are identical, so I think you should just use that instead of `mv_to_dest`